include/boost/geometry/policies/robustness/segment_ratio.hpp

```diff
@@ -94,6 +94,35 @@ struct equal<Type, false>
     }
 };
 
+template
+<
+    typename Type,
+    bool IsFloatingPoint = std::is_floating_point<Type>::type::value
+>
+struct possibly_collinear {};
+
+template <typename Type>
+struct possibly_collinear<Type, true>
+{
+    template <typename Ratio, typename Threshold>
+    static inline bool apply(Ratio const& ratio, Threshold threshold)
+    {
+        return std::abs(ratio.denominator()) < threshold;
+    }
+};
+
+// Any ratio based on non-floating point (integer or arithmetic)
+// will not be collinear.
```

Comment:
What do you mean by arithmetic? Do you mean user-defined types? They could represent FP number right?

Also why integers won't be possibly collinear? I mean for them possibly collinear and strictly collinear would be the same, right? My point is I don't understand why does this specialization return always `false`.

Reply:
Good point I will have a look

Reply:
:heavy_check_mark: I changed it. Thanks. I will merge it because next PR depends on it.